expr/function/into.go

```diff
@@ -0,0 +1,104 @@
+package function
+
+import (
+	"sort"
+
+	"github.com/brimdata/zed"
+	"github.com/brimdata/zed/field"
+	"github.com/brimdata/zed/zcode"
+)
+
+// https://github.com/brimdata/zed/blob/main/docs/language/functions.md#into
+type Into struct {
+	zcode.Builder
+	keyType zed.Type
+	mapper  *zed.Mapper
+	zctx    *zed.Context
+}
+
+func NewInto(zctx *zed.Context) *Into {
+	return &Into{
+		mapper: zed.NewMapper(zctx),
+		zctx:   zctx,
+	}
+}
+
+func (n *Into) Call(ctx zed.Allocator, args []zed.Value) *zed.Value {
+	val := args[0]
+	typ := zed.TypeRecordOf(val.Type)
+	if typ == nil {
+		return &val
+	}
+	valType := n.mapper.Lookup(typ.ID())
+	if valType == nil {
+		types := collectTypes(typ.Columns)
+		types = dedupeTypes(types)
+		if len(types) == 1 {
+			valType = types[0]
+		} else {
+			valType = n.zctx.LookupTypeUnion(types)
+		}
+		n.mapper.EnterType(typ.ID(), valType)
+	}
+	if n.keyType == nil {
+		n.keyType = n.zctx.LookupTypeArray(zed.TypeString)
+	}
+	n.Reset()
+	n.encode(typ.Columns, valType, field.Path{}, val.Bytes)
+	return ctx.NewValue(n.zctx.LookupTypeMap(n.keyType, valType), n.Bytes()).Copy()
+}
+
+func dedupeTypes(types []zed.Type) []zed.Type {
+	sort.SliceStable(types, func(i, j int) bool {
+		return zed.CompareTypes(types[i], types[j]) < 0
+	})
+	out := make([]zed.Type, 0, len(types))
+	var prev zed.Type
+	for _, typ := range types {
+		if typ != prev {
+			out = append(out, typ)
+		}
+		prev = typ
+	}
+	return out
+}
+
+func (n *Into) encode(cols []zed.Column, inner zed.Type, base field.Path, b zcode.Bytes) {
+	iter := b.Iter()
+	for _, col := range cols {
+		val := iter.Next()
```

Comment:
At this point, nearly all of our iterator variables are called `it`.
```suggestion
	it := b.Iter()
	for _, col := range cols {
		val := it.Next()
```